xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -90,8 +91,18 @@ type xdsClientInterface interface {
 
 type clusterImplBalancer struct {
 	balancer.ClientConn
-	bOpts  balancer.BuildOptions
+
+	// mu protects closed and the parent ClientConn. It's to make sure the run()
```

Comment:
This `mu` is used more to guarantee mutual exclusion between `Close()` and handling of picker update in `run()`. It is less about protecting access to `closed` and the `parent ClientConn`, right? Could we make that obvious in the comment. It feels a little odd to me if we don't mention that because `grpcsync.Event` is supposed to be safe for concurrent access.

Reply:
Updated